libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -742,6 +749,70 @@ void RC_Channel::do_aux_function_camera_trigger(const AuxSwitchPos ch_flag)
         camera->take_picture();
     }
 }
+
+void RC_Channel::do_aux_function_record_video(const AuxSwitchPos ch_flag)
+{
+    AP_Camera *camera = AP::camera();
+    if (camera == nullptr) {
+        return;
+    }
+    camera->record_video(ch_flag == AuxSwitchPos::HIGH);
+}
+
+void RC_Channel::do_aux_function_camera_zoom(const AuxSwitchPos ch_flag)
+{
+    AP_Camera *camera = AP::camera();
+    if (camera == nullptr) {
+        return;
+    }
+    int8_t zoom_step = 0;   // zoom out = -1, hold = 0, zoom in = 1
+    switch (ch_flag) {
+    case AuxSwitchPos::HIGH:
+        zoom_step = 1;  // zoom in
+        break;
+    case AuxSwitchPos::MIDDLE:
+        zoom_step = 0;  // zoom hold
+        break;
+    case AuxSwitchPos::LOW:
+        zoom_step = -1; // zoom out
+        break;
+    }
+    camera->set_zoom_step(zoom_step);
+}
+
+void RC_Channel::do_aux_function_camera_manual_focus(const AuxSwitchPos ch_flag)
+{
+    AP_Camera *camera = AP::camera();
+    if (camera == nullptr) {
+        return;
+    }
+    int8_t focus_step = 0;  // focus in = -1, focus hold = 0, focus out = 1
+    switch (ch_flag) {
+    case AuxSwitchPos::HIGH:
+        // wide shot, focus out
+        focus_step = 1;
+        break;
+    case AuxSwitchPos::MIDDLE:
+        focus_step = 0;
+        break;
+    case AuxSwitchPos::LOW:
+        // close shot, focus in
+        focus_step = -1;
+        break;
+    }
+    camera->set_manual_focus_step(focus_step);
+}
+
+void RC_Channel::do_aux_function_camera_auto_focus(const AuxSwitchPos ch_flag)
+{
+    AP_Camera *camera = AP::camera();
+    if (camera == nullptr) {
+        return;
+    }
+    if (ch_flag == AuxSwitchPos::HIGH) {
+        camera->set_auto_focus();
```

Comment:
Can this not be unset?  Or is this a one-shot thing?  In which case `do_auto_focus()` might be more appropriate

Reply:
@peterbarker,

I'm not completely sure if it is a one-time thing or whether this enables auto focus from this point on.  If it is the latter then I suspect that setting manual focus ton focus_hold will freeze the focus.